control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -495,6 +498,9 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		case kubeAPIServerCondition != nil && kubeAPIServerCondition.Status == metav1.ConditionFalse:
 			reason = kubeAPIServerCondition.Reason
 			message = kubeAPIServerCondition.Message
+		case !kasExternalDNSHostnameReachable:
```

Comment:
The assumption that the management cluster has full egress access for this healtch check to succeed might not always be valid.
We need a way express that.

The health check should have the authority to break the Available condition only if egress is possible. I.e:
1 - Check egress.
2 - Check hostname resolves appropriately (or build a kube client from the admin kubeconfig).

If 1 fails then this does not interfere in the "Available" and we set another condition externalDNSReachable = unknown,  reason=egressFailed message="management cluster egress does not allow check external dns reachability"

If 1 succeeds and 2 fail this interferes in the available condition as in the PR now.

Makes sense? cc @csrwng

Reply:
Are we talking here about public internet access? Couldn't we test that by pining am arbitrary hostname that we know is always available e.g. `google.com` ?